sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+6. kubeadm starts the kubelet process via systemd.
+7. Kubelet then creates a certificate signing request and sends it to the
+   workload kubernetes API server, using the service account token as the
+   authorisation header. The serving certificate of the control plane is checked
+   beforehand against that provided in the configuration.
+8. Kubeadm *on the worker node* checks for the existence of the CSR on the
+   workload API server and  *authorises* the CSR request. The only check,
+   performed on the client side is that the machine name is not already
+   registered.
+9. Workload control plane's running instance of the
+   kubernetes-controller-manager signs the certificate.
+10. Kubelet retrieves the signed certificate, and then registers itself as a
+    node in the workload control plane.
+11. Bootstrap controller keeps track of the existence of the worker node in the
+    workload cluster. If the node does not appear within the expiry period (15
+    mins) of the token, the bootstrap controller will keep renewing the token
+    without any limits to how many times it can be renewed.
+12. Cloud controller manager of the workload cluster will keep track of the
+    existence of the new worker node and then validate the node name against the
+    EC2 API to confirm that node is now part of a cluster and does not need to
+    be deleted.
+
+### 3rd Party Requirements (source, libraries, services, APIs)
+
+SBOMs are not currently produced by the Cluster API project, but dependencies
+are viewable in GitHub at:
+
+* [https://github.com/kubernetes-sigs/cluster-api/network/dependencies](https://github.com/kubernetes-sigs/cluster-api/network/dependencies)
+* [https://github.com/kubernetes-sigs/cluster-api-provider-aws/network/dependencies](https://github.com/kubernetes-sigs/cluster-api-provider-aws/network/dependencies)
+* [https://github.com/kubernetes-sigs/cluster-api-provider-azure/network/dependencies](https://github.com/kubernetes-sigs/cluster-api-provider-azure/network/dependencies)
+* [https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/network/dependencies](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/network/dependencies)
+
+For all the subprojects, the main dependency is
+[Kubernetes Controller Runtime](https://github.com/kubernetes-sigs/controller-runtime)
+, and for each of the providers, it would be there primary infrastructure API,
+e.g. AWS SDK for Go, Microsoft Azure SDK for Go and
+[govmomi](https://github.com/vmware/govmomi).
+
+Cluster API requires a bootstrap cluster in order to produce a self-managed
+management cluster, and we normally recommend
+“[kind](https://kind.sigs.k8s.io/)” (Kubernetes in Docker), which depends on
+Docker or Podman for operation.
+
+#### Identified Gaps
+
+A process should be implemented to generate an SBOM for the primary artifacts of
+the Cluster API subprojects, including OCI images and command line binaries.
```

Comment:
Qs: (sorry if some of them are silly, but 'm learning a lot here)
is there a specific format for SBOM?
what is the expected usage of an SBOM? 
Is there tooling to automatically produce a SBOM?
Do we have an issue in CAPI to track this GAP? if yes, let's link

Reply:
I think we'll both learn a lot by following folks on that issue: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/3325 :)

Reply:
Since this is covered in STRIDE-TAMPER-1, it makes sense to remove this section in favor of brevity!

Reply:
Thx for the information. Sounds good tome